java/languages.antlr/src/org/netbeans/modules/languages/antlr/v4/Antlr4ParserResult.java

```diff
@@ -211,7 +211,7 @@ protected ParseTreeListener createStructureListener() {
 
             @Override
             public void exitLexerRuleSpec(ANTLRv4Parser.LexerRuleSpecContext ctx) {
-                if (ctx.FRAGMENT() == null) {
+                if ((ctx.FRAGMENT() == null) && (ctx.TOKEN_REF() != null)) {
```

Comment:
This reminds me of a question I had: I understand, that fragments don't end up in the parse tree, but from a user perspective, I'm very interested in them (they are after all primitives I can use to structure my grammar), so it would be good to show fragments in the structure tree.

Reply:
True.